Bots/JavaScript/Skills/CodeFirst/WaterfallSkillBot/dialogs/activityRouterDialog.js

```diff
@@ -29,24 +29,24 @@ const UPDATE_DIALOG = 'Update';
  */
 class ActivityRouterDialog extends ComponentDialog {
   /**
-   * @param {string} serverUrl
+   * @param {import('../config').DefaultConfig} configuration
    * @param {import('botbuilder').ConversationState} conversationState
-   * @param {import('../skillConversationIdFactory').SkillConversationIdFactory} conversationIdFactory
+   * @param {import('botbuilder').SkillConversationIdFactory} conversationIdFactory
    * @param {import('botbuilder').SkillHttpClient} skillClient
    * @param {Object<string, import('./proactive/continuationParameters').ContinuationParameters>} continuationParametersStore
    */
-  constructor (serverUrl, conversationState, conversationIdFactory, skillClient, continuationParametersStore) {
+  constructor (configuration, conversationState, conversationIdFactory, skillClient, continuationParametersStore) {
     super(MAIN_DIALOG);
 
-    this.addDialog(new CardDialog(CARDS_DIALOG, serverUrl))
-      .addDialog(new WaitForProactiveDialog(PROACTIVE_DIALOG, serverUrl, continuationParametersStore))
-      .addDialog(new MessageWithAttachmentDialog(ATTACHMENT_DIALOG, serverUrl))
-      .addDialog(new AuthDialog(AUTH_DIALOG, process.env))
-      .addDialog(new SsoSkillDialog(SSO_DIALOG, process.env))
+    this.addDialog(new CardDialog(CARDS_DIALOG, configuration))
```

Comment:
Can we still just pass `configuration.serverUrl`, passing the entire config is convenient but it obscures the contract of the class. It makes it harder to identify the impact of configuration changes

Reply:
Feedback applied, thanks @gabog